include/mega/node.h

```diff
@@ -67,8 +69,16 @@ struct MEGA_API NewNode : public NodeCore
 
     bool added;
 
+    MEGA_DISABLE_COPY(NewNode)
+
+    NewNode(NewNode&& o);
+    NewNode& operator=(NewNode&& o);
```

Comment:
NewNode's only owned member is fileattributes, so you can do the same thing here and just make that one a unique_pointer which automatically disables copy and allows the default move constructor/assign to work. (rule of 0).  No need to write and maintain explicit move/assign when the default is correct anyway.